src/index.js

```diff
@@ -19,35 +19,35 @@ export default function loader(content) {
   });
 
   let outputPath = '';
+  const filePath = this.resourcePath;
 
   if (options.outputPath) {
     // support functions as outputPath to generate them dynamically
     outputPath = (
-      typeof options.outputPath === 'function' ? options.outputPath(url) : options.outputPath
+      typeof options.outputPath === 'function' ? options.outputPath(url, filePath) : options.outputPath
     );
   }
 
-  const filePath = this.resourcePath;
-
   if (options.useRelativePath) {
     const issuerContext = (this._module && this._module.issuer
       && this._module.issuer.context) || context;
 
     const relativeUrl = issuerContext && path.relative(issuerContext, filePath).split(path.sep).join('/');
 
     const relativePath = relativeUrl && `${path.dirname(relativeUrl)}/`;
-    // eslint-disable-next-line no-bitwise
-    if (~relativePath.indexOf('../')) {
-      outputPath = path.posix.join(outputPath, relativePath, url);
-    } else {
-      outputPath = relativePath + url;
+
+    // do not override outputPath if it's set
```

Comment:
Is this a breaking change, can you please provide a before/after example to clearify?

Reply:
I don't have a before sample, I just use this loader these days.

For the after sample, Please see my webpack config. I want to set both the `name` and the `outputPath`